### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by PROMETHIA-27**
Converted (probably) the entire codebase to use proper localization. Requires thorough testing to ensure no regressions, and preferably a code review as well

### CODE REVIEW

File: src/main/resources/assets/reignofnether/lang/en_us.json [hunk 1]
```diff
@@ -1,42 +1,677 @@
...
+    "abilities.reignofnether.promote_illager.tooltip1": "\uE004  %ds  \uE005  ",
+    "abilities.reignofnether.promote_illager.tooltip2": "Promote an illager to a captain, giving it a banner that gives",
+    "abilities.reignofnether.promote_illager.tooltip3": "a speed buff to all friendly units in a %d block radius.",
+    "abilities.reignofnether.promote_illager.tooltip4": "You may only have one captain at a time per upgraded castle.",
+    "abilities.reignofnether.roar": "Roar",
+    "abilities.reignofnether.roar.tooltip1": "\uE006  %d   \uE004  %ds  \uE005   ",
+    "abilities.reignofnether.roar.tooltip2": "Emit a deafening roar, knocking away, damaging",
+    "abilities.reignofnether.roar.tooltip3": "and briefly slowing down all nearby enemies. ",
+    "abilities.reignofnether.evoker_fangs_circular": "Evoker Fangs (Circular)",
+    "abilities.reignofnether.evoker_fangs_circular.tooltip1": "\\uE006  %d  \\uE004  %ds  \\uE005  ",
```
Comment (NumberSir) [inline src/main/resources/assets/reignofnether/lang/en_us.json]: `"abilities.reignofnether.evoker_fangs_circular.tooltip1": "\uE006  %d  \uE004  %ds  \uE005  ",`

File: src/main/resources/assets/reignofnether/lang/en_us.json [hunk 2]
```diff
@@ -1,42 +1,677 @@
...
+    "abilities.reignofnether.healing_potion.tooltip2": "Throw a potion that restores health to units.",
+    "abilities.reignofnether.lingering_harming_potion": "Lingering Harming Potion",
+    "abilities.reignofnether.lingering_harming_potion.tooltip1": "\uE006  3  \uE004  %ds  \uE005  ",
+    "abilities.reignofnether.lingering_harming_potion.tooltip2": "Throw a potion that leaves a cloud of deadly gas.",
+    "abilities.reignofnether.lingering_healing_potion": "Lingering Healing Potion",
+    "abilities.reignofnether.lingering_healing_potion.tooltip1": "\uE007  3  \uE004  %ds  \uE005  ",
+    "abilities.reignofnether.lingering_healing_potion.tooltip2": "Throw a potion that leaves a cloud of healing vapours.",
+    "abilities.reignofnether.shield_stance": "Shield Stance",
+    "abilities.reignofnether.shield_stance.tooltip1": "Raise or lower a shield - reducing projectile ",
+    "abilities.reignofnether.shield_stance.tooltip2": "damage taken by 67%% and movement speed by 50%%.",
```
Comment (NumberSir) [inline src/main/resources/assets/reignofnether/lang/en_us.json:111]: `"abilities.reignofnether.shield_stance.tooltip2": "damage taken by 67% and movement speed by 50%.",`

File: src/main/resources/assets/reignofnether/lang/en_us.json [hunk 3]
```diff
@@ -1,42 +1,677 @@
...
+    "research.reignofnether.advanced_portals": "Advanced Portals",
+    "research.reignofnether.advanced_portals.tooltip1": "Allows portals to be constructed off of nether terrain",
+    "research.reignofnether.advanced_portals.tooltip2": "and enables them to be upgraded into transport portals,",
+    "research.reignofnether.advanced_portals.tooltip3": "which can instantly teleport units between them.",
+    "research.reignofnether.blaze_firewall": "Blaze Firewall",
+    "research.reignofnether.blaze_firewall.tooltip1": "Allows Blazes to fire a single shot that ignites all",
+    "research.reignofnether.blaze_firewall.tooltip2": "blocks in a line, but disables their attacks temporarily.",
+    "research.reignofnether.brute_shields": "Brute Shields",
+    "research.reignofnether.brute_shields.tooltip1": "Allows Brutes to raise a shield to reduce projectile ",
+    "research.reignofnether.brute_shields.tooltip2": "damage taken by 67%% and movement speed by 50%%.",
```
Comment (NumberSir) [inline src/main/resources/assets/reignofnether/lang/en_us.json:370]: `"research.reignofnether.brute_shields.tooltip2": "damage taken by 67% and movement speed by 50%.",`

File: src/main/resources/assets/reignofnether/lang/en_us.json [hunk 4]
```diff
@@ -1,42 +1,677 @@
...
+    "units.piglins.reignofnether.ghast.tooltip3": "Slow and fragile, but can only be targeted by ranged units.",
+    "units.piglins.reignofnether.ghast.tooltip4": "Requires a Fortress.",
+    "units.piglins.reignofnether.grunt": "Grunt",
+    "units.piglins.reignofnether.grunt.tooltip1": "A piglin worker that can construct and ",
+    "units.piglins.reignofnether.grunt.tooltip2": "repair buildings and gather resources.",
+    "units.piglins.reignofnether.headhunter": "Headhunter",
+    "units.piglins.reignofnether.head_hunter.tooltip1": "A piglin that throws tridents in battle.",
+    "units.piglins.reignofnether.head_hunter.tooltip2": "Requires a Bastion.",
+    "units.piglins.reignofnether.hoglin": "Hoglin",
+    "units.piglins.reignofnether.hoglin.tooltip1": "A beast trained for war. Deals 50%% more damage to buildings.",
```
Comment (NumberSir) [inline src/main/resources/assets/reignofnether/lang/en_us.json:662]: `"units.piglins.reignofnether.hoglin.tooltip1": "A beast trained for war. Deals 50% more damage to buildings.",`

**Review by NumberSir — COMMENTED** [review]
typos: `\\` & `%%`

File: src/main/resources/assets/reignofnether/lang/en_us.json (hunk 3 above)
Comment (PROMETHIA-27) [reply src/main/resources/assets/reignofnether/lang/en_us.json:370]: This one's intentional, localizations are formatted and % must be escaped

File: src/main/resources/assets/reignofnether/lang/en_us.json (hunk 1 above)
Comment (PROMETHIA-27) [reply src/main/resources/assets/reignofnether/lang/en_us.json]: Thanks! I missed that

**Comment by SoLegendary** [comment]
Sorry, I'm about to give you some more in 1.0.6

**Comment by Mafuyu33** [comment]
I don't know why, but I can't run this branch. I translated and proofread a Simplified Chinese zh_cn file based on the format of the en_us.json file, hoping it can be of help.
[zh_cn.json](https://github.com/user-attachments/files/17536453/zh_cn.json)

**Comment by NumberSir** [comment]
> I don't know why, but I can't run this branch. I translated and proofread a Simplified Chinese zh_cn file based on the format of the en_us.json file, hoping it can be of help. [zh_cn.json](https://github.com/user-attachments/files/17536453/zh_cn.json)

~~咦？是b站的马夫鱼吗？~~ 

简体中文的版本我们已经翻译好一份啦，是在 https://github.com/SoLegendary/reignofnether/pull/50 和 https://github.com/SoLegendary/reignofnether/pull/51 的基础上翻译的：https://github.com/NumberSir/reignofnether/blob/localization/src/main/resources/assets/reignofnether/lang/zh_cn.json

**Comment by Mafuyu33** [comment]
> > I don't know why, but I can't run this branch. I translated and proofread a Simplified Chinese zh_cn file based on the format of the en_us.json file, hoping it can be of help. [zh_cn.json](https://github.com/user-attachments/files/17536453/zh_cn.json)
> 
> ~咦？是b站的马夫鱼吗？~
> 
> 简体中文的版本我们已经翻译好一份啦，是在 #50 和 #51 的基础上翻译的：https://github.com/maaaxinfinity/reignofnether/blob/localization/src/main/resources/assets/reignofnether/lang/zh_cn.json

真是一场酣畅淋漓的重复造轮子/(ㄒoㄒ)/~~

**Comment by KlparetlR** [comment]
是这样的，劳动重复了，这边写了vp补丁汉化的作者(小喵)本来也打算搞国际化的，被劝回去了()

**Comment by PROMETHIA-27** [comment]
This PR now compiles and runs (to the main menu at least) without issue. Tests would be very appreciated to try and break things/find missing localizations. Will be testing it when i get the chance.

**Comment by NumberSir** [comment]
@PROMETHIA-27 names in research completed message needed to be localized: [ResearchClient#L31](https://github.com/PROMETHIA-27/reignofnether/blob/localization/src/main/java/com/solegendary/reignofnether/research/ResearchClient.java#L31)

**Comment by Small-Miao** [comment]
@PROMETHIA-27  About Unit Name And Builidng Name I18n on class package com.solegendary.reignofnether.hud.HudClientEvents

UnitName Use this function getSimpleEntityName

**Pull request merged**